### PR TITLE
mcs: add store and region interface in scheduling server

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve?

Issue Number: Ref https://github.com/tikv/pd/issues/5839

### What is changed and how does it work?

```commit-message
```

1. move store and region response to another pkg
2. add store and region interface in the scheduling server

### Check List

Tests

- Unit test
- Manual test
image
image
image
image

### Release note

```release-note
None.
```

### CODE REVIEW

File: pkg/response/store.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+const (
+	// DisconnectedName is the name when store is disconnected.
+	DisconnectedName = "Disconnected"
+	// DownStateName is the name when store is down.
+	DownStateName = "Down"
+)
+
+// BuildStoreInfoResp builds a storeInfo response.
+func BuildStoreInfoResp(opt *sc.ScheduleConfig, store *core.StoreInfo) *StoreInfo {
```
Comment (rleungx) [inline pkg/response/store.go:91]: How about just `BuildStoreInfo`?

**Review by rleungx — APPROVED** [review]

File: pkg/mcs/scheduling/server/apis/v1/api.go [hunk 2]
```diff
@@ -174,9 +177,19 @@ func (s *Service) RegisterOperatorsRouter() {
...
+	router := s.root.Group("stores")
+	router.GET("", getAllStores)
+	router.GET("/:id", getStoreByID)
+}
+
 // RegisterRegionsRouter registers the router of the regions handler.
 func (s *Service) RegisterRegionsRouter() {
 	router := s.root.Group("regions")
+	router.GET("", getAllRegions)
+	router.GET("/:id", getRegionByID)
```
Comment (CabinfeverB) [inline pkg/mcs/scheduling/server/apis/v1/api.go:191]: Will we need to support GetRegionByKey in the future? If so, I think it's best to change the path to be consistent with api server?

Comment (lhy1024) [reply pkg/mcs/scheduling/server/apis/v1/api.go:191]: I think the interface may not be supported. The main purpose of this pr is to determine if the region and store are successfully synchronized between the api server and the scheduling server. cc @rleungx

Comment (CabinfeverB) [reply pkg/mcs/scheduling/server/apis/v1/api.go:191]: rest LGTM

**Review by CabinfeverB — APPROVED** [review]

File: pkg/mcs/scheduling/server/apis/v1/api.go (hunk 2 above)
Comment (rleungx) [reply pkg/mcs/scheduling/server/apis/v1/api.go:191]: The main purpose of this PR is to debug. If we do need it later, we can add it.

**Pull request merged**